src/porepy/models/contact_mechanics_biot_model.py

```diff
@@ -412,7 +412,35 @@ def _reference_scalar(self, sd: pp.Grid) -> np.ndarray:
         """
         return np.zeros(sd.num_cells)
 
+    def _compressibility(self, sd: pp.Grid) -> Union[float, np.ndarray]:
+        """Set unitary compressibility (also called storativity or storage coefficient).
+
+        Args:
+            sd: Subdomain grid.
+
+        Returns:
+            Unitary compressibility. If AD is used, an np.ndarray of ones of shape
+                (sd.num_cells, ) is returned. Otherwise, 1.0 is returned.
+
+        """
+
+        if self._use_ad:
```

Comment:
I don't think this is needed. The biot coefficient is a special case for some annoying reason which I don't remember atm. It should be safe to return the array both with and without ad.

Reply:
Ok. I will refactor this to return only an `np.ndarray`.